chemprop/utils.py

```diff
@@ -528,3 +528,86 @@ def save_smiles_splits(data_path: str,
+    :param missing_to_defaults: Whether to replace missing training arguments with the current defaults for :class: `~chemprop.args.TrainArgs`.
+        This is used for backwards compatibility.
+    :param for_molecule_fingerprint: Indicates whether this argument update is being used for molecule fingerprint generation,
+        in which case molecule features are not needed, even if used in the model training.
+    """
+    for key, value in vars(train_args).items():
+        if not hasattr(predict_args, key):
+            setattr(predict_args, key, value)
+
+    if missing_to_defaults:
+        # If a default argument would cause different behavior than occurred in legacy checkpoints before the argument existed,
+        # then that argument must be included in the `override_defaults` dictionary to force the legacy behavior.
+        override_defaults = dict((
+            ('bond_features_scaling',False),
+            ('no_bond_features_scaling',True),
+            ('atom_descriptors_scaling',False),
+            ('no_atom_descriptors_scaling',True),
+        ))
+        default_train_args=TrainArgs().parse_args(['--data_path',None,'--dataset_type',str(train_args.dataset_type)])
+        for key, value in vars(default_train_args).items():
+            if not hasattr(predict_args,key):
+                if key in override_defaults.keys():
+                    setattr(predict_args,key,override_defaults[key])
+                else:
+                    setattr(predict_args, key, value)
+    
+    # Same number of molecules must be used in training as in making predictions
+    if train_args.number_of_molecules != predict_args.number_of_molecules:
+        raise ValueError('A different number of molecules was used in training '
+                        f'model than is specified for prediction, {train_args.number_of_molecules} '
+                         'smiles fields must be provided')
+
+    # If atom-descriptors were used during training, they must be used when predicting and vice-versa
+    if train_args.atom_descriptors != predict_args.atom_descriptors:
+        raise ValueError('The use of atom descriptors is inconsistent between training and prediction. If atom descriptors '
+                         ' were used during training, they must be specified again during prediction using the same type of '
+                         ' descriptors as before. If they were not used during training, they cannot be specified during prediction.')
+
+    # If bond features were used during training, they must be used when predicting and vice-versa
+    if (train_args.bond_features_path is None) != (predict_args.bond_features_path is None):
+        raise ValueError('The use of bond descriptors is different between training and prediction. If you used bond'
+                         'descriptors for training, please specify a path to new bond descriptors for prediction.')
```

Comment:
Needs a space between bond and descriptors. Similar for the features_scaling error message below.